include/sdf/ParserConfig.hh

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef SDF_PARSER_CONFIG_HH_
+#define SDF_PARSER_CONFIG_HH_
+
+#include <functional>
+#include <map>
+#include <string>
+#include <vector>
+
+#include <sdf/Error.hh>
+#include "sdf/sdf_config.h"
+#include "sdf/system_util.hh"
+
+namespace sdf
+{
+inline namespace SDF_VERSION_NAMESPACE
+{
+// Forward declare private data class.
+class ParserConfigPrivate;
+
+/// This class contains configuration options for the libsdformat parser.
```

Comment:
In addition to below comment (how path resolution works), it's unclear to me what the precedence is for these different attributes (e.g. does callback override search paths, or the other way around).

Is there documentation that this can forward-reference, or should it be written here?

Reply:
I added some documentation in SDFImpl.hh (https://github.com/osrf/sdformat/pull/439/commits/6f53a9f113b1d2b8e4749ee5a55f76913b9f162b) and referenced it here in https://github.com/osrf/sdformat/pull/439/commits/1d321e01b8a1b5befb6097c4eac7c512b0c769f4

Reply:
Yay! Thanks!